backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -186,11 +185,27 @@ func setSnapshot(snapshotCache cacheWrapper, rtdsLayerName string, cluster strin
 	// No experiments meaning clear all experiments for the given upstream cluster
 	if len(experiments) != 0 {
 		for _, experiment := range experiments {
-			testConfig := &serverexperimentation.TestConfig{}
-			if !isFaultTest(experiment, testConfig) {
+			httpFaultConfig := &serverexperimentation.HTTPFaultConfig{}
+			if !isFaultTest(experiment, httpFaultConfig) {
```

Comment:
while you're in here wdyt about changing `isFaultTest` to be something like `maybeUnmarshalFaultTest(experiment) (*serverexperimentation.HttpFaultConfig, bool)`? The function name reads like a predicate but it actually mutates the input params which seems a bit odd